plugins/operators/__init__.py

```diff
@@ -1416,6 +1416,202 @@ def execute(self, ctx):
         ctx.dataset.create_index(field_name, unique=unique, wait=False)
 
 
+class ConvertIndexToUnique(foo.Operator):
+    @property
+    def config(self):
+        return foo.OperatorConfig(
+            name="convert_index_to_unique",  # Must match what's in fiftyone.yml
+            label="Convert an existing index to unique",
+            description="Convert an existing non-unique index to a unique index in the dataset if possible or identify any unique key violations that would occur.",
+            allow_immediate_execution=True,
+            allow_delegated_execution=True,
+            default_choice_to_delegated=False,
+        )
+
+    def resolve_input(self, ctx):
+        import re
+
+        inputs = types.Object()
+        indexes = ctx.dataset.get_index_information()
+
+        if not indexes:
+            inputs.str(
+                "index_name",
+                label="This dataset has no non-unique indexes to update",
+                view=types.Warning(),
+            )
+            return types.Property(
+                inputs,
+                invalid=True,
+            )
+
+        indexes = [
+            i
+            for i, v in indexes.items()
+            if not re.match(r"\.?id$", i) and "unique" not in v
+        ]
+
+        index_selector = types.DropdownView()
+        for key in indexes:
+            index_selector.add_choice(key, label=key)
+
+        inputs.str(
+            "index_name",
+            label="Index",
+            description="The existing index to make unique",
+            view=index_selector,
+            required=True,
+        )
+        inputs.bool(
+            "dry_run",
+            required=False,
+            default=False,
+            label="Check for unique key violations only (dry run)",
```

Comment:
I would recommend making the description a bit more verbose so that users fully understand what happens when this is checked vs not checked.

For example (note that `label` may contain markdown):

```py
label="**(Dry run)**: check whether the field contains unique values without modifying the index
```